src/bit-fields-macros/src/lib.rs

```diff
@@ -0,0 +1,1070 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Do not use this. Use [bit-fields](https://docs.rs/bit-fields).
+#![warn(clippy::pedantic, clippy::restriction)]
+#![allow(
+    clippy::non_ascii_literal,
+    clippy::blanket_clippy_restriction_lints,
+    clippy::implicit_return,
+    clippy::items_after_statements
+)]
+use std::collections::HashSet;
+use std::convert::TryFrom;
+use std::fmt::{self, Write};
+
+use proc_macro2::{Delimiter, Group, TokenStream, TokenTree};
```

Comment:
It is difficult to avoid, to use `quote!` requires `proc_macro2::TokenStream`. To avoid this would be a significant change and make the code worse. `proc_macro2` is a mainstay crate of the Rust ecosystem and I do not think we should have much aversions to using it here.

Reply:
I was not suggesting to remove it (I actually think it helps with testing). The comment was more from an aesthetically driven direction since you are importing `proc_macro2::{...,TokenStream}` but in the code below you use `proc_macro2::TokenStream`. You can just use TokenStream alone.

Reply:
Ahh, okay, yes.